packages/mdc-top-app-bar/short/foundation.ts

```diff
@@ -46,16 +46,13 @@ export class MDCShortTopAppBarFoundation extends MDCTopAppBarBaseFoundation {
     }
 
     if (!this.adapter_.hasClass(cssClasses.SHORT_COLLAPSED_CLASS)) {
-      this.scrollHandler_ = () => this.shortAppBarScrollHandler_();
-      this.adapter_.registerScrollHandler(this.scrollHandler_);
+      this.handleScroll = () => this.shortAppBarScrollHandler_();
```

Comment:
This logic needs to be moved to `handleScroll` callback itself and exit early when short collapsed class is found.

```ts
handleScroll() {
  if (this.adapter_.hasClass(cssClasses.SHORT_COLLAPSED_CLASS)) {
    return;
  }
  // ...
}
```

this.handleScroll = () => this.shortAppBarScrollHandler_